contracts/src/v0.7/KeeperRegistry.sol

```diff
@@ -726,11 +726,13 @@ contract KeeperRegistry is
     gasUsed = gasUsed - gasleft();
 
     uint96 payment = calculatePaymentAmount(gasUsed, params.adjustedGasWei, params.linkEth);
-    upkeep.balance = upkeep.balance.sub(payment);
-    upkeep.lastKeeper = params.from;
-    s_upkeep[params.id] = upkeep;
-    uint96 newBalance = s_keeperInfo[params.from].balance.add(payment);
-    s_keeperInfo[params.from].balance = newBalance;
+
+    uint96 newUpkeepBalance = s_upkeep[params.id].balance.sub(payment);
+    s_upkeep[params.id].balance = newUpkeepBalance;
```

Comment:
Nice, this saves gas by only setting individual parameters on the struct in storage instead of setting the whole struct.

Before: `78_439`
After: `77_130`

Makes me think there's more optimization we could do if we gas-golfed the whole contract.

Reply:
Nice, how do you calculate the before and after gas? 

The main motivation for this change is to read the balance from storage to allow updating balance in performUpkeep, the overall gas saving is a nice side effect.

Reply:
After I noticed the storage access changed here, I added a local measurement in one of the tests in `KeeperRegistry.test.ts` and printed it out in both `develop` and this branch.

Something like this:
```
const tx = await registry.performUpkeep(...)
const receipt = await tx.wait()
console.log(recept.gasUsed.toString())
```

Reply:
Spent a bit of time more time on it, couldn't find any obvious optimisations.